### PR TITLE
Fix scale invariance for both perspective and orthographic projections

### PR DESCRIPTION
**Pull request by hhsaez**
## Summary

This PR fixes several issues for `<GLText />` when scale invariance is enabled:

### Orthographic projection

Scale invariance is not working correctly when using orthographic projection (`perspective = false` in camera state) because we're using the w-component to compute the text scale. But that component is always 1 in orthographic mode. So, in that case we need a different approach. Check the comments in shaders for more information.

### Pixel size

Rendered text's size was not matching the pixel size specified in scaleInvariantFontSize. Now it does, although the calculations are different for orthographic and perspective projections.

### Marker's scale

After some discussion with the team, we decided to ignore the scale property of individual markers when scale invariance is enabled, since that resulted in text being too big or too small. It is taken into account when not using scale invariance, of course.

### Use Regl Context instead of WorldviewReactContext

I'm using Regl Context values for things like viewport dimensions and pixel sizes, which seem to be a more correct approach than using WorldviewReactContext. I also added a couple of properties for the camera.

## Test plan

Added new tests for orthographic projection and ignoring individual markers' scale when scale invariance is enabled.

## Versioning impact

Patch. No new functionality has been added.

### CODE REVIEW

File: packages/regl-worldview/src/camera/camera.js [hunk 1]
```diff
@@ -78,6 +78,14 @@ export default (regl: any) => {
...
           return selectors.billboardRotation(this.cameraState);
         },
+
+        isPerspective(context, props) {
+          return this.cameraState.perspective;
+        },
+
+        fovy(context, props) {
+          return this.cameraState.fovy;
+        },
```
Comment (troygibb) [inline packages/regl-worldview/src/camera/camera.js]: Just thinking out loud... presumably we could get these values from the `projection` value above, since that is derived from `cameraState`, is that correct? Not that we would want to do it that way.

It's curious to me that we haven't had to pass down these values before. I guess we haven't had to manipulate marker props so directly before (`scale` in this case).

File: packages/regl-worldview/src/commands/GLText.js [hunk 2]
```diff
@@ -228,12 +258,12 @@ const frag = `
     // depth test don't work together nicely for partially-transparent pixels.
     float edgeStep = smoothstep(1.0 - cutoff - fwidth(dist), 1.0 - cutoff, dist);
 
-    if (scaleInvariant && vBillboard == 1.0 && scaleInvariantSize < 0.03) {
-      // If scale invariant is enabled and scaleInvariantSize is "too small", do not interpolate 
-      // the raw distance value since at such small scale, the SDF approach causes some 
+    if (scaleInvariant && vBillboard == 1.0 && scaleInvariantSize <= 20.0) {
```
Comment (troygibb) [inline packages/regl-worldview/src/commands/GLText.js]: this is a pretty big bump from 0.03-- why the change?

**Comment by hhsaez** [comment]
> Is that a regression?

Yes, sorry. I was using pixel ratio which is not the correct way and that's why there were so many differences in the tests. I fixed that in my latest commit.

File: packages/regl-worldview/src/camera/camera.js (hunk 1 above)
Comment (hhsaez) [reply packages/regl-worldview/src/camera/camera.js]: Indeed, I haven't seen those properties being used anywhere else. GLText does a lot of things differently than other commands like Cubes or Lines. 
While `fovy` can be derived from the perspective projection matrix, it does requires a bit math and I prefer not to make shaders more complex than what they are right now. `isPerspective` cannot be derived, though.

File: packages/regl-worldview/src/commands/GLText.js (hunk 2 above)
Comment (hhsaez) [reply packages/regl-worldview/src/commands/GLText.js]: It's higher because this new approach does not normalize `scaleInvariantSize` by dividing it by the screen height before sending it to the shaders. The end end result is the same as before, to avoid using SDF if the character size is too small (less than 20 pixels).

File: packages/regl-worldview/src/stories/GLText.stories.js [hunk 3]
```diff
@@ -18,13 +18,22 @@ function textMarkers({
...
   background?: ?boolean,
+  randomScale?: ?boolean,
 }) {
   const radius = 10;
   const count = 10;
+  const scale = (i: number) => {
+    if (!randomScale) {
+      return { x: 1, y: 1, z: 1 };
+    }
+    // It's not really random. Otherwise, screenshot tests won't work correctly
```
Comment (janpaul123) [inline packages/regl-worldview/src/stories/GLText.stories.js]: You could also use the `rng` helper function which uses a fixed seed in screenshot tests 😄 https://github.com/cruise-automation/webviz/blob/94302209e46a6787216ad9aea2ed1d8f9cc69d1c/packages/regl-worldview/src/stories/util.js#L24

Comment (hhsaez) [reply packages/regl-worldview/src/stories/GLText.stories.js]: That's great! Thanks!
